pennylane/spin/spin_hamiltonian.py

```diff
@@ -93,4 +96,187 @@ def transverse_ising(
+        )
+
+    hamiltonian = 0.0 * qml.I(0)
+    if coupling.shape == (neighbour_order, 3):
+        for edge in lattice.edges:
+            i, j, order = edge[0], edge[1], edge[2]
+            hamiltonian += (
+                coupling[order][0] * (X(i) @ X(j))
+                + coupling[order][1] * (Y(i) @ Y(j))
+                + coupling[order][2] * (Z(i) @ Z(j))
+            )
+    else:
+        for edge in lattice.edges:
+            i, j = edge[0], edge[1]
+            hamiltonian += (
+                coupling[0][i][j] * X(i) @ X(j)
+                + coupling[1][i][j] * Y(i) @ Y(j)
+                + coupling[2][i][j] * Z(i) @ Z(j)
+            )
+
+    return hamiltonian.simplify()
+
+
+def fermihubbard(
+    lattice,
+    n_cells,
+    hopping=None,
+    interaction=1.0,
+    boundary_condition=False,
+    neighbour_order=1,
+    mapping="jordan_wigner",
+):
+    r"""Generates the Hubbard model on a lattice.
+
+    The Hamiltonian is represented as:
+
+    .. math::
+
+        \hat{H} = -t\sum_{<i,j>, \sigma}(c_{i\sigma}^{\dagger}c_{j\sigma}) + U\sum_{i}n_{i \uparrow} n_{i\downarrow}
+
+    where ``t`` is the hopping term representing the kinetic energy of electrons, and ``U`` is the on-site Coulomb interaction,
+    representing the repulsion between electrons.
+    This function assumes there are two fermions with opposite spins on each site.
+
+    Args:
+       lattice (str): Shape of the lattice. Input Values can be ``'chain'``, ``'square'``,
+                      ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, or ``'kagome'``.
+       n_cells (List[int]): Number of cells in each direction of the grid.
+       hopping (float or List[float] or List[math.array(float)]):Hopping interaction between spins, it can be a
+                      list of length equal to ``neighbour_order`` or a square matrix of size
+                      ``(num_spins, num_spins)``. Default value is [1.0].
+       interaction (float or List[float]): Coulomb interaction between spins, it can be a constant or a list of length ``num_spins``.
+       boundary_condition (bool or list[bool]): Defines boundary conditions different lattice axes,
+           default is ``False`` indicating open boundary condition.
+       neighbour_order (int): Specifies the interaction level for neighbors within the lattice.
+                       Default is 1 (nearest neighbour).
+       mapping (str): Specifies the fermion-to-qubit mapping. Input values can be
+                      ``'jordan_wigner'``, ``'parity'`` or ``'bravyi_kitaev'``.
+
+    Returns:
+       pennylane.operator: Hamiltonian for the Fermi-Hubbard model.
+
+    **Example**
+
+    >>> n_cells = [2,2]
+    >>> h = [[0.5]]
+    >>> u = [[1.0]]
+    >>> spin_ham = fermihubbard("Square", n_cells, hopping=h, interaction=u)
+    >>> print(spin_ham)
+
+    """
+
+    lattice = _generate_lattice(lattice, n_cells, boundary_condition, neighbour_order)
+
+    if hopping is None:
```

Comment:
Docstring says "Default value is [1.0].", please check.

Reply:
I have changed the default to 1.0. It was a thing with pylint, it won't let me pass [1.0] as default, and initially the code didn't take a number, now I have added the logic for code to accept a number so it should work.